sky/provision/aws/instance.py

```diff
@@ -791,3 +793,19 @@ def get_cluster_info(region: str,
         instances=instances,
         head_instance_id=head_instance_id,
     )
+
+
+def query_ports(
+    cluster_name_on_cloud: str,
+    ip: str,
```

Comment:
These two arguments seem a little bit redundant... Normally we only need one of them at the same time and we could retrieve the other one using the given one. Just brainstorming, can the returned endpoint not contain the URL, but instead have a 'format' function that takes an IP and outputs an endpoint? That way, we don't need the IP argument here.

Reply:
Fixed in https://github.com/skypilot-org/skypilot/pull/2713/commits/db9fb6b96e5047e34fee26426c6d77f4eaf33ed4